Repository: plone.app.event


Branch: refs/heads/3.2.x
Date: 2021-07-05T21:55:33+02:00
Author: Maurits van Rees (mauritsvanrees) <maurits@vanrees.org>
Commit: https://github.com/plone/plone.app.event/commit/1c13394ec55f30f643dd13118b03848656db025a

Do not allow file: protocol in ical url.

Previously, only `file://` was disallowed, but this left room for relative paths.
Taken over from [PloneHotfix20210518](https://plone.org/security/hotfix/20210518/server-side-request-forgery-via-event-ical-url).

Files changed:
A news/3274.bugfix
M plone/app/event/ical/importer.py
M plone/app/event/tests/test_ical_import.py

b'diff --git a/news/3274.bugfix b/news/3274.bugfix\nnew file mode 100644\nindex 00000000..70d2bf13\n--- /dev/null\n+++ b/news/3274.bugfix\n@@ -0,0 +1,4 @@\n+Do not allow ``file:`` protocol in ical url.\n+Previously, only ``file://`` was disallowed, but this left room for relative paths.\n+Taken over from `PloneHotfix20210518 <https://plone.org/security/hotfix/20210518/server-side-request-forgery-via-event-ical-url>`_.\n+[maurits]\ndiff --git a/plone/app/event/ical/importer.py b/plone/app/event/ical/importer.py\nindex e5520b3e..a6eeb71c 100644\n--- a/plone/app/event/ical/importer.py\n+++ b/plone/app/event/ical/importer.py\n@@ -229,8 +229,8 @@ def no_file_protocol_url(value):\n \n     This opens up security issues.\n     """\n-    if value and value.startswith("file://"):\n-        raise Invalid(_(u"URLs with file:// are not allowed."))\n+    if value and value.startswith("file:"):\n+        raise Invalid(_(u"URLs with file: are not allowed."))\n     return True\n \n \ndiff --git a/plone/app/event/tests/test_ical_import.py b/plone/app/event/tests/test_ical_import.py\nindex dffd9af9..16cadaa1 100644\n--- a/plone/app/event/tests/test_ical_import.py\n+++ b/plone/app/event/tests/test_ical_import.py\n@@ -73,3 +73,20 @@ def test_enable_ical_import(self):\n         self.assertIn(\n             \'URL to an external icalendar resource file\',\n             self.browser.contents)\n+\n+    def test_constraint(self):\n+        self.portal.invokeFactory("Folder", "f1")\n+        f1 = self.portal["f1"]\n+        f1_url = f1.absolute_url()\n+        transaction.commit()\n+\n+        # Enable ical import.\n+        self.browser.open(f1_url + "/ical_import_settings/enable")\n+        self.browser.getControl("Confirm action").click()\n+\n+        # Set it to a file url.\n+        self.browser.open(f1_url + "/ical_import_settings")\n+        self.assertIn("URL to an external icalendar resource file", self.browser.contents)\n+        self.browser.getControl(name="form.widgets.ical_url").value = "file:///tmp/test.ical"\n+        self.browser.getControl(name="form.buttons.save").click()\n+        self.assertIn("URLs with file: are not allowed.", self.browser.contents)\n'

Repository: plone.app.event


Branch: refs/heads/3.2.x
Date: 2021-07-08T09:38:16+02:00
Author: Maurits van Rees (mauritsvanrees) <m.van.rees@zestsoftware.nl>
Commit: https://github.com/plone/plone.app.event/commit/1ab5ad21da54fe23b6d31de43eaa1710e747cf8b

Merge pull request #335 from plone/hotfix-20210518-32x

Do not allow file: protocol in ical url. [3.2.x]

Files changed:
A news/3274.bugfix
M plone/app/event/ical/importer.py
M plone/app/event/tests/test_ical_import.py

b'diff --git a/news/3274.bugfix b/news/3274.bugfix\nnew file mode 100644\nindex 00000000..70d2bf13\n--- /dev/null\n+++ b/news/3274.bugfix\n@@ -0,0 +1,4 @@\n+Do not allow ``file:`` protocol in ical url.\n+Previously, only ``file://`` was disallowed, but this left room for relative paths.\n+Taken over from `PloneHotfix20210518 <https://plone.org/security/hotfix/20210518/server-side-request-forgery-via-event-ical-url>`_.\n+[maurits]\ndiff --git a/plone/app/event/ical/importer.py b/plone/app/event/ical/importer.py\nindex e5520b3e..a6eeb71c 100644\n--- a/plone/app/event/ical/importer.py\n+++ b/plone/app/event/ical/importer.py\n@@ -229,8 +229,8 @@ def no_file_protocol_url(value):\n \n     This opens up security issues.\n     """\n-    if value and value.startswith("file://"):\n-        raise Invalid(_(u"URLs with file:// are not allowed."))\n+    if value and value.startswith("file:"):\n+        raise Invalid(_(u"URLs with file: are not allowed."))\n     return True\n \n \ndiff --git a/plone/app/event/tests/test_ical_import.py b/plone/app/event/tests/test_ical_import.py\nindex dffd9af9..16cadaa1 100644\n--- a/plone/app/event/tests/test_ical_import.py\n+++ b/plone/app/event/tests/test_ical_import.py\n@@ -73,3 +73,20 @@ def test_enable_ical_import(self):\n         self.assertIn(\n             \'URL to an external icalendar resource file\',\n             self.browser.contents)\n+\n+    def test_constraint(self):\n+        self.portal.invokeFactory("Folder", "f1")\n+        f1 = self.portal["f1"]\n+        f1_url = f1.absolute_url()\n+        transaction.commit()\n+\n+        # Enable ical import.\n+        self.browser.open(f1_url + "/ical_import_settings/enable")\n+        self.browser.getControl("Confirm action").click()\n+\n+        # Set it to a file url.\n+        self.browser.open(f1_url + "/ical_import_settings")\n+        self.assertIn("URL to an external icalendar resource file", self.browser.contents)\n+        self.browser.getControl(name="form.widgets.ical_url").value = "file:///tmp/test.ical"\n+        self.browser.getControl(name="form.buttons.save").click()\n+        self.assertIn("URLs with file: are not allowed.", self.browser.contents)\n'

